### PR TITLE
kafka use-case guide

### PR DESCRIPTION
**Pull request by hairyhum**

### CODE REVIEW

File: documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md [hunk 1]
```diff
@@ -1,6 +1,136 @@
-# End-to-end encrypted messaging with Ockam and Kafka
+# End-to-end encrypted messaging with Kafka and Ockam Secure Channels
+
+## Background
+
+When using Apache Kafka and other stream processing services a common concern is data access and security.
+
+A lot of fields like finance and personal information handling require or advise data to be encrypted when stored and transferred.
```
Comment (antoinevg) [inline documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md:7]: Many fields, such as finance and personal information handling, have statutory requirements for data to be encrypted when stored or in transit.

File: documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md [hunk 2]
```diff
@@ -1,6 +1,136 @@
...
+
+Since modern networks and data pipelines involve multiple steps when data is transferred or stored, it becomes more apparent that just encoding data in transit between two endpoints or while in storage does not provide enough security.
```
Comment (antoinevg) [inline documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md:9]: Modern networks and data pipelines transfer data through multiple endpoints, often controlled by multiple vendors. It is therefore not sufficient for encryption to only cover the transit between arbitrary pairs of endpoints or once data comes to rest in storage.

File: documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md [hunk 3]
```diff
@@ -1,6 +1,136 @@
...
+
+This leads to emerging of End-to-End encryption as a preferred model for messaging.
+
+There were several attempts to define and implement end-to-end encryption for messaging system like Kafka, for example [KIP-317](https://cwiki.apache.org/confluence/display/KAFKA/KIP-317%3A+Add+end-to-end+data+encryption+functionality+to+Apache+Kafka)
+
+Many of those attemots are using some sort of a key storage and exchanging persistent encryption keys between devices. This means that the data is available to any party able to aquire the key.
```
Comment (antoinevg) [inline documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md:15]: Most of these attempts are using some sort of key storage and exchange persistent keys between devices. A fundamental weakness of such a system is its vulnerability to any party able to acquire the key.

File: documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md [hunk 4]
```diff
@@ -1,6 +1,136 @@
...
+
+Every message you type is encrypted by Secure Channel, sent to the Hub Node and put in the Kafka topic by the Hub Node.
+
+Then it's fetched by the other program and decrypted by Secure Channel.
+
+The messages printed as `Push to the stream` and `Pull from the stream` are the messages sent to Kafka topics.
+
+Messages in those logs are the only information about the messages that a network ever sees and cannot be decrypted by anyone other then the programs.
+
+**NOTE** encryption key is transient and generated on secure channel establishment. After you exit the example programs the messages stored in Kafka will be essentially lost.
```
Comment (antoinevg) [inline documentation/guides/draft/use-cases/end-to-end-encryption-with-kafka/README.md:124]: s // will be essentially lost. // will be lost as they can no longer be decrypted.

**Review by antoinevg — COMMENTED** [review]
Just some small changes to phrasing here and there.

**Comment by hairyhum** [comment]
merged to #1503

**Pull request closed without merging**